### PR TITLE
HBASE-22526 RejectedExecutionException could be thrown from TableOver…

### PR DESCRIPTION
**Pull request by Apache9**
…AsyncTable.coprocessor service if the connection has been shutown

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 186 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ HBASE-21512 Compile Tests _ |
| +1 | mvninstall | 250 | HBASE-21512 passed |
| +1 | compile | 21 | HBASE-21512 passed |
| +1 | checkstyle | 27 | HBASE-21512 passed |
| +1 | shadedjars | 289 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 56 | HBASE-21512 passed |
| +1 | javadoc | 20 | HBASE-21512 passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 271 | the patch passed |
| +1 | compile | 22 | the patch passed |
| +1 | javac | 22 | the patch passed |
| +1 | checkstyle | 27 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 272 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 805 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | findbugs | 63 | the patch passed |
| +1 | javadoc | 18 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 97 | hbase-client in the patch passed. |
| +1 | asflicense | 11 | The patch does not generate ASF License warnings. |
| | | 2751 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-286/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/286 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux 561c986aea82 4.4.0-137-generic #163-Ubuntu SMP Mon Sep 24 13:14:43 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | HBASE-21512 / bb4d11275a |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-286/1/testReport/ |
| Max. process+thread count | 276 (vs. ulimit of 10000) |
| modules | C: hbase-client U: hbase-client |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-286/1/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

File: hbase-client/src/main/java/org/apache/hadoop/hbase/client/TableOverAsyncTable.java [hunk 1]
```diff
@@ -422,20 +425,29 @@ public RegionCoprocessorRpcChannel coprocessorService(byte[] row) {
...
-      futures.put(r, future);
+        });
+        futures.put(r, future);
+      }
+    } catch (RejectedExecutionException e) {
+      // maybe the connection has been closed, let's check
+      if (pool.isShutdown()) {
+        throw new DoNotRetryIOException("Connection is closed");
+      } else {
+        throw new HBaseIOException("Connection is closed");
```
Comment (infraio) [inline hbase-client/src/main/java/org/apache/hadoop/hbase/client/TableOverAsyncTable.java]: Why here's error message is "Connection is closed", too?

Comment (Apache9) [reply hbase-client/src/main/java/org/apache/hadoop/hbase/client/TableOverAsyncTable.java]: Fixed.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 56 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ HBASE-21512 Compile Tests _ |
| +1 | mvninstall | 256 | HBASE-21512 passed |
| +1 | compile | 23 | HBASE-21512 passed |
| +1 | checkstyle | 30 | HBASE-21512 passed |
| +1 | shadedjars | 285 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 53 | HBASE-21512 passed |
| +1 | javadoc | 20 | HBASE-21512 passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 246 | the patch passed |
| +1 | compile | 24 | the patch passed |
| +1 | javac | 24 | the patch passed |
| +1 | checkstyle | 28 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 269 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 826 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | findbugs | 65 | the patch passed |
| +1 | javadoc | 20 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 103 | hbase-client in the patch passed. |
| +1 | asflicense | 10 | The patch does not generate ASF License warnings. |
| | | 2633 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-286/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/286 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux 6131afcfa43a 4.4.0-137-generic #163-Ubuntu SMP Mon Sep 24 13:14:43 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | HBASE-21512 / bb4d11275a |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-286/2/testReport/ |
| Max. process+thread count | 297 (vs. ulimit of 10000) |
| modules | C: hbase-client U: hbase-client |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-286/2/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Pull request merged**